### ISSUE TITLE
[Feature request] more clearly set initial search paramters

### ISSUE THREAD

**Issue by higumachan** [optuna/optuna#417]
I want to search with human designed hyper parameters.

Now published solution is a [here](https://qiita.com/tmitani/items/c0acd198788089786e29).
But that solution is not clearly I think.

In my opinion is optuna should have API for search fixed parameters.

If this idea does not violate the idea of ​​optuna I would like to implement that feature.

I'm waiting for your opinions.

thanks.

**Comment by sile** [comment]
Hi,

Your proposed feature seems very useful.
Actually, we are planning to develop a feature to register trials in advance as the following:
```python
# Define objective function.
def objective(trial):
    x = trial.suggest_uniform('x', -10, 10)
    return (x - 2) ** 2

# Create a study instance.
study = optuna.create_study()

# Register the parameters that will be used in the first trial.
#
# NOTE: The API signature is under consideration and may be changed in the future.
first_trial_params = {
    'x': {
        'value': 2,
        'distribution': optuna.distributions.UniformDistribution(-10, 10)
    }
}
study.register_trial(params = first_trial_params)

# Register the parameters that will be used in the second trial.
second_trial_params = {
    'x': {
        'value': 5,
        'distribution': optuna.distributions.UniformDistribution(-10, 10)
    }
}
study.register_trial(params = second_trial_params)

# Run optimization.
#
# The first two trials will use the parameters registered via the above API calls.
study.optimize(objective)
```

I think that this API can satisfy your requirement.
(if not, please let me know your idea.)

> I would like to implement that feature.

Thank you!
However, it seems not so easy to implement this feature correctly with distributed optimization in mind, so we're going to discuss the design and implementation details within Optuna dev team.
We'll create a PR about this feature in the near feature. Your feedback would be quite helpful, then.

**Comment by higumachan** [comment]
Your API idea is very good!

I think that it is good for your idea, but I will post my first idea for the time being.

```
# Define objective function.
def objective(trial):
    x = trial.suggest_uniform('x', -10, 10)
    return (x - 2) ** 2

conclusive_sampler = ConclusiveSampler([
    {'x': {
        'value': 2,
        'distribution': optuna.distributions.UniformDistribution(-10, 10)
    }},
    {'x': {
        'value': 5,
        'distribution': optuna.distributions.UniformDistribution(-10, 10)
    }},
])

# Create a study instance.
study = optuna.create_study(sampler=CombineSampler(conclusive_sampler, TPESampler()))

# Run optimization.
#
# The first two trials will use the parameters registered via the above API calls.
study.optimize(objective)
```

> Thank you!
> However, it seems not so easy to implement this feature correctly with distributed optimization in mind, so we're going to discuss the design and implementation details within Optuna dev team.
We'll create a PR about this feature in the near feature. Your feedback would be quite helpful, then.

OK, I'm waiting your implementation! thanks.

**Comment by sile** [comment]
Thank you for sharing your idea!

**Comment by sile** [comment]
Closing this issue, since we have merged #520 that implements the `Study.enqueue_trial()` method, which can be used to set initial search parameters.
